lib/std/crypto/25519/curve25519.zig

```diff
@@ -0,0 +1,155 @@
+    }
+
+    /// Reject the neutral element.
+    pub fn rejectIdentity(p: Curve25519) !void {
+        if (p.x.isZero()) {
+            return error.IdentityElement;
+        }
+    }
+
+    fn ladder(p: Curve25519, s: [32]u8, comptime bits: usize) !Curve25519 {
+        var x1 = p.x;
+        var x2 = Fe.one();
+        var z2 = Fe.zero();
+        var x3 = x1;
+        var z3 = Fe.one();
+        var swap: u8 = 0;
+        var pos: usize = bits - 1;
+        while (true) {
+            const b = (s[pos / 8] >> @intCast(u3, pos & 7)) & 1;
+            swap ^= b;
+            Fe.cSwap2(&x2, &x3, &z2, &z3, swap);
+            swap = b;
+            var tmp0 = x3.sub(z3);
+            var tmp1 = x2.sub(z2);
+            x2 = x2.add(z2);
+            z2 = x3.add(z3);
+            z3 = tmp0.mul(x2);
+            z2 = z2.mul(tmp1);
+            tmp0 = tmp1.sq();
+            tmp1 = x2.sq();
+            x3 = z3.add(z2);
+            z2 = z3.sub(z2);
+            x2 = tmp1.mul(tmp0);
+            tmp1 = tmp1.sub(tmp0);
+            z2 = z2.sq();
+            z3 = tmp1.mul32(121666);
+            x3 = x3.sq();
+            tmp0 = tmp0.add(z3);
+            z3 = x1.mul(z2);
+            z2 = tmp1.mul(tmp0);
+            if (pos == 0) break;
+            pos -= 1;
+        }
+        Fe.cSwap2(&x2, &x3, &z2, &z3, swap);
+        z2 = z2.invert();
+        x2 = x2.mul(z2);
+        if (x2.isZero()) {
+            return error.IdentityElement;
+        }
+        return Curve25519 { .x = x2 };
+    }
+
+    /// Multiply a Curve25519 point by a scalar after "clamping" it.
+    /// Clamping forces the scalar to be a multiple of the cofactor in
+    /// order to prevent small subgroups attacks. This is the standard
+    /// way to use Curve25519 for a DH operation.
+    /// Return error.IdentityElement if the resulting point is
+    /// the identity element.
+    pub fn clampedMul(p: Curve25519, s: [32]u8) !Curve25519 {
+        var t: [32]u8 = s;
+        scalar.clamp(&t);
+        return ladder(p, t, 255);
+    }
+
+    /// Multiply a Curve25519 point by a scalar without clamping it.
+    /// Return error.IdentityElement if the resulting point is
+    /// the identity element or error.WeakPublicKey if the public
+    /// key is a low-order point.
+    pub fn mul(p: Curve25519, s: [32]u8) !Curve25519 {
+        const cofactor = [_]u8{8} ++ [_]u8{0} ** 31;
+        _ = ladder(p, cofactor, 4) catch |_| return error.WeakPublicKey;
+        return ladder(p, s, 256);
+    }
+};
+
+test "curve25519" {
+    var s = [32]u8{ 1, 2, 3, 4, 5, 6, 7, 8, 1, 2, 3, 4, 5, 6, 7, 8, 1, 2, 3, 4, 5, 6, 7, 8, 1, 2, 3, 4, 5, 6, 7, 8 };
+    const p = try Curve25519.basePoint().clampedMul(s);
+    try p.rejectIdentity();
+    var buf: [128]u8 = undefined;
+    const alloc = &std.heap.FixedBufferAllocator.init(&buf).allocator;
+    std.testing.expectEqualStrings(try std.fmt.allocPrint(alloc, "{X}", .{p.toBytes()}), "E6F2A4D1C28EE5C7AD0329268255A468AD407D2672824C0C0EB30EA6EF450145");
```

Comment:
use `std.fmt.bufPrint`

Alternatively, convert the hex to bytes using `std.fmt.hexToBytes`

Alternatively, there is a crypto/test.zig that has the function `assertEqual` that takes hex.

Reply:
Done, thank you!